src/main/java/com/marafone/marafone/game/active/ActiveGameController.java

```diff
@@ -32,11 +33,23 @@ public List<Game> getPublicGames(){
         return activeGameService.getPublicGames();
     }
 
+    // get all games to display them to user at landing page as lobbies
+    // it does same thing as getPublicGames() but transforms it into GameDTO
+    @GetMapping("/game/waiting")
+    @ResponseBody
+    public List<GameDTO> getWaitingGames() {
+        return activeGameService.getWaitingGames();
+    }
+
     @PostMapping("/game/create")
     @ResponseBody
-    public String createGame(@RequestBody CreateGameRequest createGameRequest, @AuthenticationPrincipal User user){
+    public ResponseEntity<String> createGame(@RequestBody CreateGameRequest createGameRequest, @AuthenticationPrincipal User user){
+        if (activeGameService.doesNotStartedGameAlreadyExist(createGameRequest.getGameName()))
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body("The game name already exists. Please choose a different name.");
+
         Long gameId = activeGameService.createGame(createGameRequest, user);
```

Comment:
After this if and before creating a game someone could add game with this name. Those 2 operations should be atomic. (You would have to use something like synchronize(game) to make it work).